src/main/java/org/tron/common/entity/NodeInfo.java

```diff
@@ -0,0 +1,316 @@
+  }
+
+  public NodeInfo setListenPort(int listenPort) {
+    this.listenPort = listenPort;
+    return this;
+  }
+
+  public boolean isDiscoverEnable() {
+    return discoverEnable;
+  }
+
+  public NodeInfo setDiscoverEnable(boolean discoverEnable) {
+    this.discoverEnable = discoverEnable;
+    return this;
+  }
+
+  public int getActiveNodeSize() {
+    return activeNodeSize;
+  }
+
+  public NodeInfo setActiveNodeSize(int activeNodeSize) {
+    this.activeNodeSize = activeNodeSize;
+    return this;
+  }
+
+  public int getPassiveNodeSize() {
+    return passiveNodeSize;
+  }
+
+  public NodeInfo setPassiveNodeSize(int passiveNodeSize) {
+    this.passiveNodeSize = passiveNodeSize;
+    return this;
+  }
+
+  public int getSendNodeSize() {
+    return sendNodeSize;
+  }
+
+  public NodeInfo setSendNodeSize(int sendNodeSize) {
+    this.sendNodeSize = sendNodeSize;
+    return this;
+  }
+
+  public int getMaxConnectCount() {
+    return maxConnectCount;
+  }
+
+  public NodeInfo setMaxConnectCount(int maxConnectCount) {
+    this.maxConnectCount = maxConnectCount;
+    return this;
+  }
+
+  public int getSameIpMaxConnectCount() {
+    return sameIpMaxConnectCount;
+  }
+
+  public NodeInfo setSameIpMaxConnectCount(int sameIpMaxConnectCount) {
+    this.sameIpMaxConnectCount = sameIpMaxConnectCount;
+    return this;
+  }
+
+  public int getBackupListenPort() {
+    return backupListenPort;
+  }
+
+  public NodeInfo setBackupListenPort(int backupListenPort) {
+    this.backupListenPort = backupListenPort;
+    return this;
+  }
+
+  public int getBackupMemberSize() {
+    return backupMemberSize;
+  }
+
+  public NodeInfo setBackupMemberSize(int backupMemberSize) {
+    this.backupMemberSize = backupMemberSize;
+    return this;
+  }
+
+  public int getBackupPriority() {
+    return backupPriority;
+  }
+
+  public NodeInfo setBackupPriority(int backupPriority) {
+    this.backupPriority = backupPriority;
+    return this;
+  }
+
+  public int getCurrentConnectCount() {
+    return currentConnectCount;
+  }
+
+  public NodeInfo setCurrentConnectCount(int currentConnectCount) {
+    this.currentConnectCount = currentConnectCount;
+    return this;
+  }
+
+  public int getActiveConnectCount() {
+    return activeConnectCount;
+  }
+
+  public NodeInfo setActiveConnectCount(int activeConnectCount) {
+    this.activeConnectCount = activeConnectCount;
+    return this;
+  }
+
+  public int getPassiveConnectCount() {
+    return passiveConnectCount;
+  }
+
+  public NodeInfo setPassiveConnectCount(int passiveConnectCount) {
+    this.passiveConnectCount = passiveConnectCount;
+    return this;
+  }
+
+  public long getTotalFlow() {
+    return totalFlow;
+  }
+
+  public NodeInfo setTotalFlow(long totalFlow) {
+    this.totalFlow = totalFlow;
+    return this;
+  }
+
+  public List<PeerInfo> getPeerList() {
+    return peerList;
+  }
+
+  public NodeInfo setPeerList(List<PeerInfo> peerList) {
+    this.peerList = peerList;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "peerList". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1611&issues=AWaB4VfXn3WdA7VnG1BT&open=AWaB4VfXn3WdA7VnG1BT)

[ISSUE_KEY]: AWaB4VfXn3WdA7VnG1BT